addon/index.d.ts

```diff
@@ -0,0 +1,940 @@
+
+/**
+ * A cancelation-aware variant of [RSVP.allSettled](http://emberjs.com/api/classes/RSVP.html#method_allSettled).
+ * The normal version of a `RSVP.allSettled` just returns a regular, uncancelable
+ * Promise. The `ember-concurrency` variant of `allSettled()` has the following
+ * additional behavior:
+ *
+ * - if the task that `yield`ed `allSettled()` is canceled, any of the
+ *   {@linkcode TaskInstance}s passed in to `allSettled` will be canceled
+ */
+export function allSettled<T extends readonly unknown[] | readonly [unknown]>(
+  values: T
+): Promise<{ -readonly [K in keyof T]: Settled<T[K]> }>;
+export function allSettled<T>(values: Iterable<T>): Promise<Array<Settled<T>>>;
+
+/**
+ * Returns true if the object passed to it is a TaskCancelation error.
+ * If you call `someTask.perform().catch(...)` or otherwise treat
+ * a {@linkcode TaskInstance} like a promise, you may need to
+ * handle the cancelation of a TaskInstance differently from
+ * other kinds of errors it might throw, and you can use this
+ * convenience function to distinguish cancelation from errors.
+ *
+ * ```js
+ * click() {
+ *   this.get('myTask').perform().catch(e => {
+ *     if (!didCancel(e)) { throw e; }
+ *   });
+ * }
+ * ```
+ *
+ * @param error The caught error, which might be a TaskCancelation.
+ */
+export function didCancel(error: unknown): error is TaskCancelation;
+
+/**
+ * A cancelation-aware variant of [RSVP.hash](http://emberjs.com/api/classes/RSVP.html#hash).
+ * The normal version of a `RSVP.hash` just returns a regular, uncancelable
+ * Promise. The `ember-concurrency` variant of `hash()` has the following
+ * additional behavior:
+ *
+ * - if the task that `yield`ed `hash()` is canceled, any of the
+ *   {@linkcode TaskInstance}s passed in to `allSettled` will be canceled
+ * - if any of the items rejects/cancels, all other cancelable items
+ *   (e.g. {@linkcode TaskInstance}s) will be canceled
+ */
+export function hash<T extends Record<string, unknown>>(
+  values: T
+): Promise<{ [K in keyof T]: Resolved<T[K]> }>;
+export function hash<T>(values: Record<string, T>): Promise<Record<string, Resolved<T>>>;
+
+/**
+ * A cancelation-aware variant of [Promise.race](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race).
+ * The normal version of a `Promise.race` just returns a regular, uncancelable
+ * Promise. The `ember-concurrency` variant of `race()` has the following
+ * additional behavior:
+ *
+ * - if the task that `yield`ed `race()` is canceled, any of the
+ *   {@linkcode TaskInstance}s passed in to `race` will be canceled
+ * - once any of the tasks/promises passed in complete (either success, failure,
+ *   or cancelation), any of the {@linkcode TaskInstance}s passed in will be canceled
+ *
+ * [Check out the "Awaiting Multiple Child Tasks example"](/docs/examples/joining-tasks)
+ */
+export function race<T>(values: readonly T[]): Promise<Resolved<T>>;
```

Comment:
Should this also have the extra tuple overloads?

Reply:
same